kong/plugins/key-auth/handler.lua

```diff
@@ -170,6 +172,13 @@ local function do_authentication(conf)
     return nil, ERR_NO_API_KEY
   end
 
+  -- this request contains invalid non-ascii symbol in API key, HTTP 401
+  matchposition = key:find('[\192-\255][\128-\191]*')
```

Comment:
What is the actual requirement for the API key?  Are control characters and/or DEL (\127) allowed?  Why is the pattern not '[\128-\255]*'?

Reply:
@hanshuebner as per the below documentation I found that API key is alphanumeric ([A-Z][a-z][0-9]) even though i could not find any exact requirement for it. 
https://docs.konghq.com/hub/kong-inc/key-auth/

>  Are control characters and/or DEL (\127) allowed? Why is the pattern not [\128-\255]*

As for this PR , I have used regex to not allow utf-8 characters like **à,á,â,ã,ä,å** .  I agree that [\128-\255] covers all the extended characters list. Should I make the regex like this `'[\128-\255]'`

Reply:
I'd rather whitelist what's in the documentation ("The key names may only contain [a-z], [A-Z], [0-9], [_] underscore, and [-] hyphen.")

Reply:
@hanshuebner ok I am using this regex `^[%w%-%_]+$` to allow only ` [a-z], [A-Z], [0-9], [_] underscore, and [-] hyphen` . But do you think if this could handle the scenario if the non ascii character is at the last character of the string ?

Reply:
> But do you think if this could handle the scenario if the non ascii character is at the last character of the string ?

You could add a test to check that scenario. You'd make sure your solution is working and no one will mess with it in the future.

Reply:
I misread the documentation - It is the _key name_ that must contain [a-zA-Z0-9_], but nothing is said about the key itself.  I will need to revisit.